hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,184 @@
+            return Response({}, status=404)
+
+        title = resource_file.metadata.logical_file.dataset_name \
+            if resource_file.metadata.logical_file else ""
+        keywords = resource_file.metadata.keywords \
+            if resource_file.metadata else []
+        spatial_coverage = resource_file.metadata.spatial_coverage.value \
+            if resource_file.metadata.spatial_coverage else {}
+        extra_metadata = resource_file.metadata.extra_metadata \
+            if resource_file.metadata else []
+        temporal_coverage = resource_file.metadata.temporal_coverage.value if \
+            resource_file.metadata.temporal_coverage else {}
+
+        # TODO: How to leverage serializer for this?
+        return Response({
+            "title": title,
+            "keywords": keywords,
+            "spatial_coverage": spatial_coverage,
+            "extra_metadata": extra_metadata,
+            "temporal_coverage": temporal_coverage
+        })
+
+    def put(self, request, pk, file_id):
+        """
+        Update a resource file's metadata
+
+        Accepts application/json encoding.
+
+        ## Parameters
+        * `id` - alphanumeric uuid of the resource, i.e. cde01b3898c94cdab78a2318330cf795
+        * `file_id` - integer id of the resource file. You can use the `/hsapi/resource/{id}/files`
+        to get these
+        * `data` - see the "returns" section for formatting
+
+        ## Returns
+        ```
+        {
+            "keywords": [
+                "keyword1",
+                "keyword2"
+            ],
+            "spatial_coverage": {
+                "units": "Decimal degrees",
+                "east": -84.0465,
+                "north": 49.6791,
+                "name": "12232",
+                "projection": "WGS 84 EPSG:4326"
+            },
+            "extra_metadata": {
+                "extended1": "one"
+            },
+            "temporal_coverage": {
+                "start": "2018-02-22",
+                "end": "2018-02-24"
+            },
+            "title": "File Metadata Title"
+        }
+        ```
+        """
+        file_serializer = FileMetaDataSerializer(request.data)
+
+        title = file_serializer.data.pop("title", "")
+        resource_file = ResourceFile.objects.get(id=file_id)
+        resource_file.metadata.logical_file.dataset_name = title
+        resource_file.metadata.logical_file.save()
+
+        spatial_coverage = file_serializer.data.pop("spatial_coverage", None)
+        if spatial_coverage is not None:
+            if resource_file.metadata.spatial_coverage is not None:
+                Coverage.update(resource_file.metadata.spatial_coverage.id,
+                                _value=json.dumps(spatial_coverage))
+            elif resource_file.metadata.spatial_coverage is None:
+                resource_file.metadata.create_element('coverage', type="point",
+                                                      _value=json.dumps(spatial_coverage))
```

Comment:
```value=spatial_coverage```

Reply:
I think when ```_value``` is used it bypasses the validation. I am guessing that could have resulted in bag creation failure if the json data for ```_value``` did not had the key 'start'.